superset-frontend/packages/superset-ui-chart-controls/src/types.ts

```diff
@@ -73,7 +73,7 @@ export interface Dataset {
   main_dttm_col: string;
   // eg. ['["ds", true]', 'ds [asc]']
   order_by_choices?: [string, string][] | null;
-  time_grain_sqla?: string;
+  time_grain_sqla?: [string, string][];
```

Comment:
I think this change is incorrect. If you search for `time_grain_sqla` in the codebase you'll see it's a `string` in many places.

Reply:
I think that in the context of Dataset it actually should be an array. It doesn't really make sense for a dataset to have a single value for time grain, it should rather be a list of supported time grains. Also, if you look at Redux, that's what `time_grain_sqla` is. 

<img width="345" alt="image" src="https://github.com/apache/superset/assets/15073128/e4c4e3fb-862a-4b52-9675-51d67e3fd720">

Reply:
Oh, I think you're right! I was thinking about `form_data` but that makes sense in the context of the `Dataset`. That's the problem with bad naming.